sh/drive-utils.sh

```diff
@@ -0,0 +1,535 @@
+        -T "${input_upload_file}" \
+        -o- \
+        --url "${uploadlink_upload_file}" \
+        --globoff \
+        ${CURL_SPEED} ${resume_args_upload_file} || :)"
+    return 0
+}
+
+# logging in case of successful upload
+_normal_logging() {
+    [ -z "${VERBOSE_PROGRESS}" ] && for _ in 1 2 3; do _clear_line 1; done
+    "${QUIET:-_print_center}" "justify" "${slug_upload_file} " "| ${readable_size_upload_file} | ${STRING}" "="
+    return 0
+}
+
+# Tempfile Used for resuming interrupted uploads
+_log_upload_session() {
+    [ "${inputsize_upload_file}" -gt 1000000 ] && printf "%s\n" "${uploadlink_upload_file}" >| "${__file_upload_file}"
+    return 0
+}
+
+# remove upload session
+_remove_upload_session() {
+    rm -f "${__file_upload_file}"
+    return 0
+}
+
+# wrapper to fully upload a file from scratch
+_full_upload() {
+    _generate_upload_link || { _error_logging "${slug_upload_file}" && return 1; }
+    _log_upload_session
+    _upload_file_from_uri
+    _collect_file_info "${upload_body_upload_file}" "${slug_upload_file}" || return 1
+    _normal_logging
+    _remove_upload_session
+    return 0
+}
+# Sub functions for _upload_file function - End
+###################################################
+
+###################################################
+# Upload ( Create/Update ) files on gdrive.
+# Interrupted uploads can be resumed.
+# Globals: 7 variables, 10 functions
+#   Variables - API_URL, API_VERSION, QUIET, VERBOSE, VERBOSE_PROGRESS, CURL_PROGRESS, LOG_FILE_ID
+#   Functions - _url_encode, _json_value, _print_center, _bytes_to_human
+#               _generate_upload_link, _upload_file_from_uri, _log_upload_session, _remove_upload_session
+#               _full_upload, _collect_file_info
+# Arguments: 5
+#   ${1} = update or upload ( upload type )
+#   ${2} = file to upload
+#   ${3} = root dir id for file
+#   ${4} = Access Token
+# Result: On
+#   Success - Upload/Update file and export FILE_ID
+#   Error - return 1
+# Reference:
+#   https://developers.google.com/drive/api/v3/create-file
+#   https://developers.google.com/drive/api/v3/manage-uploads
+#   https://developers.google.com/drive/api/v3/reference/files/update
+###################################################
+_upload_file() {
+    [ $# -lt 4 ] && printf "Missing arguments\n" && return 1
+    job_upload_file="${1}" input_upload_file="${2}" folder_id_upload_file="${3}" token_upload_file="${4}"
+    unset slug_upload_file inputname_upload_file extension_upload_file inputsize_upload_file readable_size_upload_file request_method_upload_file \
+        url_upload_file postdata_upload_file uploadlink_upload_file upload_body_upload_file mime_type_upload_file resume_args_upload_file
+
+    slug_upload_file="${input_upload_file##*/}"
+    inputname_upload_file="${slug_upload_file%.*}"
+    extension_upload_file="${slug_upload_file##*.}"
+    inputsize_upload_file="$(($(wc -c < "${input_upload_file}")))" && content_length_upload_file="${inputsize}"
```

Comment:
Codacy found an issue: [inputsize is referenced but not assigned.](https://app.codacy.com/manual/labbots/google-drive-upload/pullRequest?prid=5865418)